android-components/components/browser/menu2/src/main/java/mozilla/components/browser/menu2/ext/PopupWindow.kt

```diff
@@ -9,96 +9,192 @@ import android.view.Gravity
-    val fitsDown = availableHeightToBottom >= containerHeight
-    val reversed = availableWidthToLeft < availableWidthToRight
-
-    // Popup window does not need a input method. This avoids keyboard flicker when menu is opened.
-    inputMethodMode = PopupWindow.INPUT_METHOD_NOT_NEEDED
 
-    // Try to use the preferred orientation, if doesn't fit fallback to the best fit.
-    when {
-        preferredOrientation == Orientation.DOWN && (fitsDown || forceOrientation) ->
-            showPopupWithDownOrientation(anchor, reversed)
-        preferredOrientation == Orientation.UP && (fitsUp || forceOrientation) ->
-            showPopupWithUpOrientation(anchor, containerHeight, reversed)
-        else -> {
-            showPopupWhereBestFits(
-                anchor,
-                fitsUp,
-                fitsDown,
-                reversed,
-                containerHeight,
-            )
+    val containerRecyclerView =
+        menuViewContainer.findViewById<RecyclerView>(R.id.mozac_browser_menu_recyclerView)
+    val recyclerViewHeight = containerRecyclerView.measuredHeight
+    val recyclerViewAdapter = containerRecyclerView.adapter
+
+    if (recyclerViewAdapter != null) {
+        val recyclerViewItemCount = recyclerViewAdapter.itemCount
+
+        val upMenuViewContainerHeight = getMenuViewContainerHeight(
+            recyclerViewHeight,
+            recyclerViewItemCount,
+            availableHeightToTop,
+            menuViewContainer.measuredHeight,
+        )
+        val downMenuViewContainerHeight = getMenuViewContainerHeight(
+            recyclerViewHeight,
+            recyclerViewItemCount,
+            availableHeightToBottom,
+            menuViewContainer.measuredHeight,
+        )
+
+        val fitsUp = availableHeightToTop >= upMenuViewContainerHeight
+        val fitsDown = availableHeightToBottom >= downMenuViewContainerHeight
+        val reversed = availableWidthToLeft < availableWidthToRight
+
+        // Popup window does not need a input method. This avoids keyboard flicker when menu is opened.
+        inputMethodMode = PopupWindow.INPUT_METHOD_NOT_NEEDED
+
+        // Try to use the preferred orientation, if doesn't fit fallback to the best fit.
+        when {
+            preferredOrientation == Orientation.DOWN && fitsDown -> {
+                showPopupWithDownOrientation(anchor, downMenuViewContainerHeight, reversed)
+            }
+            preferredOrientation == Orientation.UP && fitsUp -> {
+                showPopupWithUpOrientation(anchor, upMenuViewContainerHeight, reversed)
+            }
+            else -> {
+                showPopupWhereBestFits(
+                    anchor,
+                    reversed,
+                    upMenuViewContainerHeight,
+                    downMenuViewContainerHeight,
+                    fitsUp,
+                    fitsDown,
+                )
+            }
         }
+    } else {
+        Log.log(
```

Comment:
Adding logging as a starter - should this do something else to capture the error e.g. included in telemetry data?

Reply:
Is this something that you expect might happen in a corner case? Or is this just something you want to record in case that it does happen?

Reply:
Definitely a corner case - I can't think of a situation where this could happen as we always have the default menu items (which can not be removed by the user). If this does somehow happen then something very unexpected has occurred. I think by checking the conditions, we can at least provide somewhat of a graceful experience - the logging may be superfluous if it is not accessible (would we get sight of errors or would this be local to the device?)